Rewrite torch.library's documentation

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #111661
* #111660
* __->__ #111310
* #111659
* #111380

We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

zou3519 committed:
Rewrite torch.library's documentation

We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

[ghstack-poisoned]

zou3519 committed:
Update on "Rewrite torch.library's documentation"


We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

[ghstack-poisoned]

zou3519 committed:
Update on "Rewrite torch.library's documentation"


We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

[ghstack-poisoned]

zou3519 committed:
Update on "Rewrite torch.library's documentation"


We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

[ghstack-poisoned]

zou3519 committed:
Update on "Rewrite torch.library's documentation"


We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

[ghstack-poisoned]

zou3519 committed:
Update on "Rewrite torch.library's documentation"


We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

[ghstack-poisoned]

zou3519 committed:
Update on "Rewrite torch.library's documentation"


We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

[ghstack-poisoned]

zou3519 committed:
Update on "Rewrite torch.library's documentation"


We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

[ghstack-poisoned]

soulitzer commented on torch/library.py:
```diff
@@ -177,11 +177,11 @@ def define(qualname, schema, *, lib=None):
     This entrypoint defines the custom operator (the first step)
     you must then perform the second step by calling various
-    ``impl_*`` APIs.
+    ``impl_*`` APIs, like :func:`torch.library.impl_device`.
```
don't forget to update this

zou3519 committed:
Update on "Rewrite torch.library's documentation"


We mention the higher-level torch.library APIs and put the original docs
into a low-level API section.

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge
